src/Emulator/Extensions/Utilities/GDB/Commands/ReadRegisterCommand.cs

```diff
@@ -20,17 +20,34 @@ public ReadRegisterCommand(CommandsManager manager) : base(manager)
         public PacketData Execute(
             [Argument(Encoding = ArgumentAttribute.ArgumentEncoding.HexNumber)]int registerNumber)
         {
-            if(!manager.Cpu.GetRegisters().Any(x => x.Index == registerNumber))
-            {
-                return PacketData.ErrorReply(1);
+            var content = new StringBuilder();
+
+            // if register exists in emulated core return current value of this
+            if(manager.Cpu.GetRegisters().Any(x => x.Index == registerNumber)) { 
+                foreach(var b in manager.Cpu.GetRegisterUnsafe(registerNumber).GetBytes())
+                {
+                    content.AppendFormat("{0:x2}", b);
+                }
+                return new PacketData(content.ToString());
             }
 
-            var content = new StringBuilder();
-            foreach(var b in manager.Cpu.GetRegisterUnsafe(registerNumber).GetBytes())
+            // if register no exists in emulated core but is defined in a feature
+            // return a cero value with number of bytes defined in the feature
+            var selectedRegisters = manager.Cpu.GDBFeatures.SelectMany(f => f.Registers)
+                .Where(r => r.Number == registerNumber);
+
+            if(selectedRegisters != null)
```

Comment:
I believe `Where` won't return `null` if there are no results, but rather an empty iterator.

Reply:
Your observations it's correct, was changed to .Count ==1